src/lib_io.c

```diff
@@ -126,30 +126,46 @@ static int io_file_close(lua_State *L, IOFileUD *iof)
 static int io_file_readnum(lua_State *L, FILE *fp)
 {
   lua_Number d;
-  if (fscanf(fp, LUA_NUMBER_SCAN, &d) == 1) {
-    if (LJ_DUALNUM) {
-      int32_t i = lj_num2int(d);
-      if (d == (lua_Number)i && !tvismzero((cTValue *)&d)) {
-	setintV(L->top++, i);
-	return 1;
-      }
+  int rc;
+  for (;;) {
+    rc = fscanf(fp, LUA_NUMBER_SCAN, &d);
+    if (rc == 1) break;
+    if (ferror(fp)) {
+      if (errno == EINTR) { clearerr(fp); continue; }
+      return 0;
+    } else {
+      setnilV(L->top++);
+      return 0;
+    }
+  }
+  if (LJ_DUALNUM) {
+    int32_t i = lj_num2int(d);
+    if (d == (lua_Number)i && !tvismzero((cTValue *)&d)) {
+      setintV(L->top++, i);
+      return 1;
     }
-    setnumV(L->top++, d);
-    return 1;
-  } else {
-    setnilV(L->top++);
-    return 0;
   }
+  setnumV(L->top++, d);
+  return 1;
 }
 
 static int io_file_readline(lua_State *L, FILE *fp, MSize chop)
 {
-  MSize m = LUAL_BUFFERSIZE, n = 0, ok = 0;
-  char *buf;
+  MSize m = LUAL_BUFFERSIZE, n = 0, ok = 0, s = 0;
```

Comment:
I don't think using the name `s` for size values is a good idea. It looks like `string` :)

Reply:
Updated the name to `size`